src/platform/Linux/bluez/ChipDeviceScanner.cpp

```diff
@@ -112,31 +112,30 @@ void ChipDeviceScanner::Shutdown()
         },
         this);
 
-    mIsInitialized = false;
+    mCurrentState = ChipDeviceScannerState::SCANNER_UNINITIALIZED;
 }
 
 CHIP_ERROR ChipDeviceScanner::StartScan(System::Clock::Timeout timeout)
 {
+    CHIP_ERROR err              = CHIP_NO_ERROR;
+    static bool startingScanner = false; // To prevent function reentrancy
```

Comment:
In current for it does not prevent reentrancy, because you are clearing that flag just after `PlatformMgrImpl().GLibMatterContextInvokeSync`, but before `mCurrentState = ChipDeviceScannerState::SCANNER_SCANNING;`.